app/src/main/java/com/github/muellerma/prepaidbalance/work/CheckBalanceWorker.kt

```diff
@@ -105,9 +107,14 @@ class CheckBalanceWorker(
             }
         }
 
+        @SuppressLint("MissingPermission")
         fun checkBalance(context: Context, callback: (CheckResult, String?) -> Unit) {
             if (!context.hasPermissions(Manifest.permission.CALL_PHONE, Manifest.permission.READ_PHONE_STATE)) {
-                return callback(CheckResult.MISSING_PERMISSIONS, null)
+                return callback(CheckResult.MISSING_PERMISSIONS_PHONE, null)
+            }
+
+            if (!context.hasPermissions(Manifest.permission.POST_NOTIFICATIONS)) {
+                return callback(CheckResult.MISSING_PERMISSIONS_NOTIFICATION, null)
             }
```

Comment:
Missing notification permissions shouldn't break getting the balance.

Reply:
As far as I tested it, the `hasPermissions()` always returns true before Android 13

Reply:
Sure, but even on Android 13 this permission isn't required for the app to work.

Reply:
Yup, got it now..
I'm just afraid that most people won't recognize that the permission is missing.. I also didn't see it until my SIM-balance was empty..
When should the user be asked for notification permission then? And what should happen in case it is missing?

Reply:
I'd show a snackbar or alert dialog (might be too annoying though) whenever the app is opened and the permission is missing.
The notification related permission should be disabled in settings.